### PR TITLE
changed json parser for array projections to accept 'projection' and 'project'

### PR DESCRIPTION
**Pull request by snehagunta**

### CODE REVIEW

File: query-api/src/main/resources/json-schema/projection/range.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
                 "type": "integer"
             }
         },
-        "project": {
+        "projection": {
```
Comment (bserdar) [inline query-api/src/main/resources/json-schema/projection/range.json]: Accept both project and projection. Keep "project" section, and add "projection" with same contents

File: query-api/src/main/resources/json-schema/projection/match.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
         "match": {
             "$ref": "/json-schema/query/choice.json#/definitions/query"
         },
-        "project": {
+        "projection": {
```
Comment (bserdar) [inline query-api/src/main/resources/json-schema/projection/match.json]: Accept both project and projection. Keep "project" section, and add "projection" with same contents

File: query-api/src/test/resources/projection/schema-test-projection-single-match-many-project.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
         "op": "$eq",
         "rfield": "someuser"
     },
-    "project": [
+    "projection": [
```
Comment (bserdar) [inline query-api/src/test/resources/projection/schema-test-projection-single-match-many-project.json]: Accept both project and projection. Keep "project" section, and add "projection" with same contents

File: query-api/src/test/resources/projection/schema-test-projection-single-match-single-project.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
         "op": "$eq",
         "rfield": "someuser"
     },
-    "project": {
+    "projection": {
```
Comment (bserdar) [inline query-api/src/test/resources/projection/schema-test-projection-single-match-single-project.json]: Accept both project and projection. Keep "project" section, and add "projection" with same contents

**Comment by jewzaam** [comment]
@bserdar @snehagunta this change will break clients using the old "project" attribute.

**Comment by jewzaam** [comment]
To be clearer, making 'projection' required instead of 'project' will mean older client requests will not be schema compliant.

**Comment by bserdar** [comment]
lets make it not required. Sneha, could you remove "projection" from the
required list?

Separate PR pls.

**Comment by snehagunta** [comment]
@jewzaam @bserdar  Ah..I missed that detail. Will make the change.

**Comment by jewzaam** [comment]
thanks @snehagunta !

**Pull request merged**